src/sentry/snuba/spans_rpc.py

```diff
@@ -239,24 +246,28 @@ def run_top_events_timeseries_query(
     top_events = run_table_query(
         params,
         query_string,
-        groupby + y_axes,
+        raw_groupby + y_axes,
         orderby,
         0,
         limit,
         referrer,
         config,
         search_resolver=search_resolver,
     )
-    groupby_columns = [col for col in groupby if not is_function(col)]
+    # Need to change the project slug columns to project.id because timeseries requests don't take virtual_column_contexts
+    groupby_columns = [col for col in raw_groupby if not is_function(col)]
+    groupby_columns_without_project = [
+        col if col not in ["project", "project.name"] else "project.id" for col in groupby_columns
+    ]
```

Comment:
Aren't there other columns that use a virtual column context? One that comes to mind is `device.class`. Should we have a more robust system to handle this?

Reply:
:+1: I'll do that in a follow up, just trying to get this working piece by piece first